osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/WatchHighScoresCommand.cs

```diff
@@ -0,0 +1,228 @@
+
+            if (firstEntry == null)
+            {
+                Console.WriteLine("Couldn't find any scores to process");
+                return -1;
+            }
+
+            if (lastImportedLegacyScoreId >= firstEntry.score_id)
+            {
+                var entry = db.QuerySingle<ScoreProcessQueue>($"SELECT * FROM score_process_queue WHERE mode = {RulesetId} AND score_id = {lastImportedLegacyScoreId + 1}");
+
+                lastQueueId = entry.queue_id;
+                Console.WriteLine($"Continuing watch mode from last processed legacy score (score_id: {entry.score_id} queue_id: {entry.queue_id})");
+            }
+            else
+            {
+                lastQueueId = firstEntry.queue_id;
+                Console.WriteLine($"WARNING: Continuing watch mode from start of the processing queue (score_id: {firstEntry.score_id} queue_id: {firstEntry.queue_id})");
+                Console.WriteLine("This implies that a full import hasn't been run yet, you might want to run another import first to catch up.");
+                Console.WriteLine();
+                Console.WriteLine("You have 10 seconds to decide if you really want to do this..");
+                await Task.Delay(10000, cancellationToken);
+            }
+
+            if (SkipScoreProcessor)
+            {
+                elasticQueueProcessor = new ElasticQueuePusher();
+                Console.WriteLine($"Indexing to elasticsearch queue(s) {elasticQueueProcessor.ActiveQueues}");
+            }
+            else
+            {
+                scoreStatisticsQueueProcessor = new ScoreStatisticsQueueProcessor();
+                Console.WriteLine($"Pushing imported scores to redis queue {scoreStatisticsQueueProcessor.QueueName}");
+            }
+
+            if (DryRun)
+                Console.WriteLine("RUNNING IN DRY RUN MODE.");
+
+            Console.WriteLine();
+            Console.WriteLine("Starting processing in 5 seconds...");
+            await Task.Delay(5000, cancellationToken);
+
+            using (var dbMainQuery = DatabaseAccess.GetConnection())
+            {
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    HighScore[] highScores = (await dbMainQuery.QueryAsync<HighScore>($"SELECT * FROM osu.score_process_queue LEFT JOIN {highScoreTable} h USING (score_id) WHERE queue_id >= @lastQueueId AND mode = {RulesetId} ORDER BY queue_id LIMIT 1000", new
+                    {
+                        lastQueueId,
+                        RulesetId,
+                    })).ToArray();
+
+                    if (highScores.Length == 0)
+                    {
+                        Thread.Sleep(500);
+                        continue;
+                    }
+
+                    var inserter = new BatchInserter(ruleset, highScores, importLegacyPP: !SkipScoreProcessor, skipExisting: true, skipNew: false, dryRun: DryRun);
```

Comment:
seems backwards?

```suggestion
                    var inserter = new BatchInserter(ruleset, highScores, importLegacyPP: SkipScoreProcessor, skipExisting: true, skipNew: false, dryRun: DryRun);
```

if score processor is skipped, then the implication is that the pp is already calculated by the legacy pathways, in which case it should be inserted across from the legacy tables

Reply:
Luckily this was not back-to-front in the commit-to-master that is deployed 😅